<?xml version="1.0" encoding="UTF-8"?>
<testsuite name="org.springframework.context.event.AnnotationDrivenEventListenerTests" tests="36" skipped="1" failures="0" errors="0" timestamp="2019-12-09T02:20:39" hostname="DESKTOP-246NTI5" time="1.05">
  <properties/>
  <testcase name="simpleEventXmlConfig()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.017"/>
  <testcase name="eventListenerWorksWithSimpleInterfaceProxy()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.011"/>
  <testcase name="asyncProcessingAppliedWithInterfaceProxy()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.016"/>
  <testcase name="listenerWithNonMatchingPayload()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.004"/>
  <testcase name="simpleEventJavaConfig()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.005"/>
  <testcase name="monoReply()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.34"/>
  <testcase name="contextEventsAreReceived()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.004"/>
  <testcase name="simpleReply()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.005"/>
  <testcase name="arrayReply()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.005"/>
  <testcase name="methodSignatureNoEvent()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.005"/>
  <testcase name="asyncProcessingApplied()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.021"/>
  <testcase name="missingListenerBeanIgnored()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.007"/>
  <testcase name="fluxReply()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.019"/>
  <testcase name="conditionMatch()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.01"/>
  <testcase name="orderedListeners()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.006"/>
  <testcase name="listenerWithGenericApplicationEvent()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.004"/>
  <testcase name="listenableFutureReply()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.013"/>
  <testcase name="collectionReplyNullValue()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.004"/>
  <testcase name="conditionDoesNotMatch()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.008"/>
  <testcase name="asyncProcessingAppliedWithScopedProxy()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.019"/>
  <testcase name="privateMethodOnCglibProxyFails()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.008"/>
  <testcase name="eventListenerWorksWithCglibProxy()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.01"/>
  <testcase name="conditionMatchWithProxy()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.405"/>
  <testcase name="listenersReceiveEarlyEvents()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.0">
    <skipped/>
  </testcase>
  <testcase name="eventListenerWorksWithCustomScope()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.01"/>
  <testcase name="collectionReply()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.003"/>
  <testcase name="eventListenerWorksWithAnnotatedInterfaceProxy()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.006"/>
  <testcase name="metaAnnotationIsDiscovered()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.004"/>
  <testcase name="exceptionNotPropagatedWithAsync()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.017"/>
  <testcase name="listenerWithResolvableTypeEvent()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.016"/>
  <testcase name="replyWithPayload()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.01"/>
  <testcase name="nullReplyIgnored()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.005"/>
  <testcase name="listenerWithResolvableTypeEventWrongGeneric()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.011"/>
  <testcase name="completableFutureReply()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.009"/>
  <testcase name="exceptionPropagated()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.004"/>
  <testcase name="listenerWithSimplePayload()" classname="org.springframework.context.event.AnnotationDrivenEventListenerTests" time="0.003"/>
  <system-out><![CDATA[10:20:39.811 [Test worker] WARN  o.s.c.a.AnnotationConfigApplicationContext - Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanInitializationException: Failed to process @EventListener annotation on bean with name 'annotationDrivenEventListenerTests.InvalidMethodSignatureEventListener'; nested exception is java.lang.IllegalStateException: Maximum one parameter is allowed for event listener method: public void org.springframework.context.event.AnnotationDrivenEventListenerTests$InvalidMethodSignatureEventListener.cannotBeCalled(java.lang.String,java.lang.Integer)
10:20:39.933 [Test worker] WARN  o.s.c.a.AnnotationConfigApplicationContext - Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanInitializationException: Failed to process @EventListener annotation on bean with name 'annotationDrivenEventListenerTests.CglibProxyWithPrivateMethod'; nested exception is java.lang.IllegalStateException: Need to invoke method 'handleIt' found on proxy for target class 'CglibProxyWithPrivateMethod' but cannot be delegated to target bean. Switch its visibility to package or protected.
10:20:40.389 [SimpleAsyncTaskExecutor-1] ERROR o.s.a.i.SimpleAsyncUncaughtExceptionHandler - Unexpected exception occurred invoking async method: public void org.springframework.context.event.AnnotationDrivenEventListenerTests$ExceptionEventListener.handleAsync(org.springframework.context.event.test.AnotherTestEvent)
java.lang.IllegalStateException: Test exception
	at org.springframework.context.event.AnnotationDrivenEventListenerTests$ExceptionEventListener.handleAsync(AnnotationDrivenEventListenerTests.java:794) ~[test/:?]
	at org.springframework.context.event.AnnotationDrivenEventListenerTests$ExceptionEventListener$$FastClassBySpringCGLIB$$356572e3.invoke(<generated>) ~[test/:?]
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218) ~[spring-core-5.2.2.RELEASE.jar:5.2.2.RELEASE]
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:769) ~[spring-aop-5.2.2.RELEASE.jar:?]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163) ~[spring-aop-5.2.2.RELEASE.jar:?]
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:747) ~[spring-aop-5.2.2.RELEASE.jar:?]
	at org.springframework.aop.interceptor.AsyncExecutionInterceptor.lambda$invoke$0(AsyncExecutionInterceptor.java:115) ~[spring-aop-5.2.2.RELEASE.jar:5.2.2.RELEASE]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_172]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_172]
]]></system-out>
  <system-err><![CDATA[]]></system-err>
</testsuite>
